### DISCUSSION TITLE
Failed to connect gitHub from Jenkins

### DISCUSSION THREAD

**Discussion by y2003huang** [community/community#57828] [Programming Help] [answered]
### Select Topic Area

Product Feedback

### Body

I didn't have a problem to connect gitHub from Jenkins before. Since last week, the connection failed. I re-generated PAT and update the gitHub Credentials with the new token. I am still getting the following errors -

Failed to connect to repository : Command "C:\Program Files\Git\cmd\git.exe ls-remote -h -- https://github.com/agero-private/connected-vehicle-gateway.git HEAD" returned status code 128:
stdout:
stderr: fatal: Cannot prompt because user interactivity has been disabled.
fatal: Cannot prompt because user interactivity has been disabled.
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/agero-private/connected-vehicle-gateway.git/'

Could you please help? Thank you very much.

**Comment by airtower-luna** [comment]
The output

> stderr: fatal: Cannot prompt because user interactivity has been disabled.

sounds like Git doesn't have a "password" (which should be your PAT). Whatever it then sends  isn't a valid token, that's what the failure message means. I can't say why because I don't know your Jenkins setup and its credential storage, but the thing to look into is *why* Git doesn't have access to a token there.

**Comment by y2003huang** [comment]
@airtower-luna Thank you very much for helping me this issue. The following is the github credential screenshot I created in Jenkins. Could you please help me to check if any information I missed?
![image](https://github.com/community/community/assets/133034880/948600d7-5384-471c-a9ae-a151fa22c1bb)

**Comment by airtower-luna** [comment, marked as answer]
The "private key" part looks odd. The token is a secret string, like a password, not a key. Could it be that the credential *type* is wrong? I don't see the type in the screenshot, but if (e.g.) Jenkins is trying to use the token as a private SSH key it'd explain why the "password" (token) for HTTPS is empty. :thinking:

**Comment by y2003huang** [comment]
@airtower-luna You are right. The problem has been resolved. Thank you very much for your help.

**Reply by ChinaYangKun** [reply]
hi I encountered the same problem, how did you solve it?

**Reply by UATCO** [reply]
@airtower-luna Hello
I encountered a similar problem.
SSH registered in git.
Created a user with SSH.
But a similar error still appears.
Help me please.

<img width="1122" alt="Снимок экрана 2023-10-03 в 08 51 25" src="https://github.com/community/community/assets/143316033/814a62b6-607f-4f46-b32d-5fa723423d1d">

<img width="1165" alt="Снимок экрана 2023-10-03 в 08 52 09" src="https://github.com/community/community/assets/143316033/93615918-0dff-4fb0-984c-c447d02410cc">

**Reply by airtower-luna** [reply]
It seems you're trying to clone over HTTPS instead of using SSH.

**Reply by UATCO** [reply]
@airtower-luna

now he's swearing like this

<img width="1440" alt="Снимок экрана 2023-10-06 в 08 24 53" src="https://github.com/community/community/assets/143316033/9f3687ab-7bb7-484b-b169-18bdfd2a2044">

**Comment by airtower-luna** [comment]
Glad I could help! Please don't forget to mark the answer. :slightly_smiling_face:

**Comment by f33r0** [comment]
```
15:03:11  The recommended git tool is: NONE
15:03:13  using credential github-user
15:03:13  Cloning the remote Git repository
15:03:14  Cloning repository https://github.com/org/repo.git
15:03:14   > git init /var/jenkins/workspace/myworkspace # timeout=10
15:03:14  Fetching upstream changes from https://github.com/org/repo.git
15:03:14   > git --version # timeout=10
15:03:14   > git --version # 'git version 2.34.1'
15:03:14  using GIT_ASKPASS to set credentials 
15:03:14   > git fetch --tags --force --progress -- https://github.com/org/repo.git +refs/heads/*:refs/remotes/origin/* # timeout=10
15:03:15  ERROR: Error cloning remote repo 'origin'
15:03:15  hudson.plugins.git.GitException: Command "git fetch --tags --force --progress -- https://github.com/org/repo.git +refs/heads/*:refs/remotes/origin/*" returned status code 128:
15:03:15  stdout: 
15:03:15  stderr: fatal: unable to access 'https://github.com/org/repo.git/': gnutls_handshake() failed: The TLS connection was non-properly terminated.
```

Has anyone encountered such a problem? I use https to download repositories.

**Reply by Oct-Dora** [reply]
I've the same issue.
seems like something to do with how Jenkins uses the Git Token.
For example : https://{https_token}@github.com/{org_name}/{repo_name}.git
